src/lecture_feedback/thread_safe_dict.py

```diff
@@ -1,32 +1,50 @@
+from __future__ import annotations
+
 import threading
-from collections.abc import Iterator
-from typing import Any, Self
+from collections import UserDict
+from typing import TYPE_CHECKING, Any, Self
+
+if TYPE_CHECKING:
+    from collections.abc import ItemsView, Iterator
+
 
+class ThreadSafeDict(UserDict[str, Any]):
+    """A simple thread-safe dictionary with only the methods we need"""
 
-class ThreadSafeDict:
-    def __init__(self) -> None:
-        self._data: dict[str, Any] = {}
+    def __init__(self, *args: Any, **kwargs: Any) -> None:  # noqa: ANN401
+        # Set up the lock first so any operations performed by the
+        # parent `UserDict` initializer (which may call `update`) can
+        # safely use our overridden methods that expect `_lock`.
         self._lock = threading.RLock()
+        super().__init__(*args, **kwargs)
 
-    def __setitem__(self, key: str, value: Any) -> None:  # noqa: ANN401
+    def __getitem__(self, key: str) -> Any:  # noqa: ANN401
         with self._lock:
-            self._data[key] = value
+            return super().__getitem__(key)
 
-    def __getitem__(self, key: str) -> Any:  # noqa: ANN401
+    def __setitem__(self, key: str, value: Any) -> None:  # noqa: ANN401
         with self._lock:
-            return self._data[key]
+            return super().__setitem__(key, value)
 
     def __delitem__(self, key: str) -> None:
         with self._lock:
-            del self._data[key]
+            return super().__delitem__(key)
+
+    def __iter__(self) -> Iterator[str]:
+        with self._lock:
+            return iter(list(self.data))  # safe copy
```

Comment:
I think here it would be good to have comment on the function that this will invoca a copy in contrast to a normal dict